etherman/etherman.go

```diff
@@ -652,13 +652,13 @@ func (etherMan *Client) forcedBatchEvent(ctx context.Context, vLog types.Log, bl
 	forcedBatch.BlockNumber = vLog.BlockNumber
 	forcedBatch.ForcedBatchNumber = fb.ForceBatchNum
 	forcedBatch.GlobalExitRoot = fb.LastGlobalExitRoot
+
 	// Read the tx for this batch.
-	tx, isPending, err := etherMan.EthClient.TransactionByHash(ctx, vLog.TxHash)
+	tx, err := etherMan.getTxFromLog(ctx, &vLog)
 	if err != nil {
 		return err
```

Comment:
```suggestion
	tx, err := etherMan.EthClient.TransactionInBlock(ctx, vLog.BlockHash, vLog.TxIndex)
	if err != nil {
		return err
```

Reply:
As `EthClient` is an interface of `ethereumClient` 

https://github.com/0xPolygonHermez/zkevm-node/blob/317e8b304ce76c4dbb097ec94edaa891f79decc4/etherman/etherman.go#L104

Which didn't implement the `TransactionInBlock` method.

yeah you may say we can change it from interface to `ethclient.Client` struct, but NewSimulatedEtherman https://github.com/0xPolygonHermez/zkevm-node/blob/317e8b304ce76c4dbb097ec94edaa891f79decc4/etherman/simulated.go#L22 also need this field :)

Reply:
sorry, SimulatedEtherman has implemented this method.